builtin/providers/opsgenie/resource_opsgenie_team.go

```diff
@@ -0,0 +1,232 @@
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateOpsGenieTeamName,
+			},
+			"member": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"role": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "user",
+							ValidateFunc: validateOpsGenieTeamRole,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceOpsGenieTeamCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*OpsGenieClient).teams
+
+	name := d.Get("name").(string)
+
+	createRequest := team.CreateTeamRequest{
+		Name:    name,
+		Members: expandOpsGenieTeamMembers(d),
+	}
+
+	log.Printf("[INFO] Creating OpsGenie team '%s'", name)
+
+	createResponse, err := client.Create(createRequest)
+	if err != nil {
+		return err
+	}
+
+	err = checkOpsGenieResponse(createResponse.Code, createResponse.Status)
+	if err != nil {
+		return err
+	}
+
+	getRequest := team.GetTeamRequest{
+		Name: name,
+	}
+
+	getResponse, err := client.Get(getRequest)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(getResponse.Id)
+
+	return resourceOpsGenieTeamRead(d, meta)
+}
+
+func resourceOpsGenieTeamRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*OpsGenieClient).teams
+
+	listRequest := team.ListTeamsRequest{}
+	listResponse, err := client.List(listRequest)
+	if err != nil {
+		return err
+	}
+
+	var found *team.GetTeamResponse
+	for _, team := range listResponse.Teams {
+		if team.Id == d.Id() {
+			found = &team
+			break
+		}
+	}
+
+	if found == nil {
+		d.SetId("")
+		log.Printf("[INFO] Team %q not found. Removing from state", d.Get("name").(string))
+		return nil
+	}
+
+	getRequest := team.GetTeamRequest{
+		Id: d.Id(),
+	}
+
+	getResponse, err := client.Get(getRequest)
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", getResponse.Name)
+	d.Set("member", flattenOpsGenieTeamMembers(getResponse.Members))
+
+	return nil
+}
+
+func resourceOpsGenieTeamUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*OpsGenieClient).teams
+	name := d.Get("name").(string)
+
+	updateRequest := team.UpdateTeamRequest{
+		Id:      d.Id(),
+		Name:    name,
+		Members: expandOpsGenieTeamMembers(d),
+	}
+
+	log.Printf("[INFO] Updating OpsGenie team '%s'", name)
+
+	updateResponse, err := client.Update(updateRequest)
+	if err != nil {
+		return err
+	}
+
+	err = checkOpsGenieResponse(updateResponse.Code, updateResponse.Status)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceOpsGenieTeamDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Deleting OpsGenie team '%s'", d.Get("name").(string))
+	client := meta.(*OpsGenieClient).teams
+
+	deleteRequest := team.DeleteTeamRequest{
```

Comment:
Will this be allowed if there are members attached to the Team? e.g. AWS doesn't allow us to delete an iam_role if policies are attached to it